### DISCUSSION TITLE
hcl2-parser package unsupported

### DISCUSSION THREAD

**Discussion by viceice** [renovatebot/renovate#21985] [Suggest an Idea]
### Type of discussion.

I'm proposing an idea

### Tell us more.

the [hcl2-parser](https://www.npmjs.com/package/hcl2-parser) project is no longer maintained and is archived. It also has security issues because of very outdated go.

1. We should fork it and modernize.
2. We can fork [`@ts-terraform/hcl`](https://github.com/vt-cpolcini/ts-terraform/tree/master/packages/%40ts-terraform/hcl)  and modernize.
3. Use [@cdktf/hcl2json](https://www.npmjs.com/package/@cdktf/hcl2json) 🙃 

the second one uses wasm instead of converting go to js.

**Comment by JamieMagee** [comment]
Just throwing out ideas, but there is also [`tree-sitter-hcl`][1] which is maintained.

[1]: https://github.com/MichaHoffmann/tree-sitter-hcl

**Reply by secustor** [reply]
I think this only supports HCLv1 and not v2 🤔

**Reply by JamieMagee** [reply]
The example shows string interpolation, which is an HCL v2 feature.

**Comment by viceice** [comment]
- https://github.com/renovatebot/renovate/pull/21988
